napari/utils/notebook_display.py

```diff
@@ -75,7 +86,7 @@ def _repr_png_(self):
     def _repr_html_(self):
         png = self._repr_png_()
         url = 'data:image/png;base64,' + base64.b64encode(png).decode('utf-8')
-        html = f'<img src="{url}"></img>'
+        html = f'<img src="{url}" alt="{self.alt_text}"></img>'
```

Comment:
That will be incorrect if there is a `"` in `self.alt_text` (that true also for url but less likely, you likely want:

```suggestion
        html = f'<img src="{url}" alt="{html.escape(self.alt_text)}"></img>'
```

And import `html` somewhere ([stdlib](https://docs.python.org/3/library/html.html))

I think quotes, or other html signs are more likely in an alt text than other places.

https://github.com/ipython/ipython/pull/12864/files#diff-744c8a0c7792b132c2bbde8802d8c5234175720ddb921111c57b4aa2cac673bcR991 might be of help also it adds alt attributes to the core `Image`, you might be able to use similar naming for attributes (like just `alt` if you wish to, or directly use an Image object and forward the repr calls to it ?

Reply:
Another question is if the alt text is explicitly set to `None`, should the alt attribute be skipped or an empty string ? I'm tempted to think a missing alt would be better than an empty string, but I might be wrong.

Reply:
If the image is truly decorative and you don't want it to be announced by a screen reader, then it should be the empty string: https://www.w3.org/WAI/tutorials/images/decorative/

If it's missing, some screen readers may fall back to using the file name as the alt text, which might have some useful information, but will probably just be confusing.

Reply:
Hum, I'm still unsure what to do. I think that passing None to not have the `alt` attribute make sens. If we allow None we need extra logic depending on wether alt_text is None otherwise html.escape will fail

Reply:
> If the image is truly decorative

This seems unlikely for a screenshot.